bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1138,6 +1139,8 @@ private void registerAndInitializeHandler(final Thing thing,
                             identifier);
                 }
             } else {
+                setThingStatus(thing, new ThingStatusInfo(ThingStatus.UNINITIALIZED,
+                        ThingStatusDetail.HANDLER_REGISTERING_ERROR, "Handler factory not found"));
```

Comment:
Wouldn't the right sub-status be `HANDLER_MISSING_ERROR` here?
Have you checked, how often we run into this code? My hope would be that at startup we only add the things once the handlers are already available - if not, we will see one such event for every Thing on startup, which would be a bit annoying.

Reply:
Unfortunately this happens quite often (at least for managed things). This will not show up in the log, though.

I agree on the `HANDLER_MISSING_ERROR`.

Reply:
> This will not show up in the log, though.

Not even in the event log?

Reply:
No. I guess this is because it happens very early in the startup process where no events are logged.

Reply:
Ok, then I'd hope we can live with that during startup.
So could you please change the code to `HANDLER_MISSING_ERROR`? Once this is done, I can merge the PR.

Reply:
Done